### PR TITLE
drivers: can: can_loopback: Abort pending messages when stopped

### PR DESCRIPTION
**Pull request by benjaminbjornsson**
The CAN header file specifies that stopping the CAN controller should result in "aborting any pending CAN frame transmissions". This PR aims at adding that missing functionality to the CAN loopback driver.

### CODE REVIEW

**Review by alexanderwachter — DISMISSED** [review]
Looks good to me. Thanks for the fix!

File: drivers/can/can_loopback.c [hunk 1]
```diff
@@ -66,12 +66,16 @@ static void tx_thread(void *arg1, void *arg2, void *arg3)
...
+	int ret;
 
 	ARG_UNUSED(arg2);
 	ARG_UNUSED(arg3);
 
 	while (1) {
-		k_msgq_get(&data->tx_msgq, &frame, K_FOREVER);
+		ret = k_msgq_get(&data->tx_msgq, &frame, K_FOREVER);
+		if (ret < 0) {
+			continue;
```
Comment (henrikbrixandersen) [inline drivers/can/can_loopback.c:77]: Nit: add a debug log here.

**Review by henrikbrixandersen — DISMISSED** [review]

File: drivers/can/can_loopback.c (hunk 1 above)
Comment (benjaminbjornsson) [reply drivers/can/can_loopback.c:77]: Would this do: `LOG_DBG("Pend on TX queue returned without valid frame (err %d)", ret);`?

Comment (henrikbrixandersen) [reply drivers/can/can_loopback.c:77]: Something along those lines would be fine, thanks.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by str4t0m — APPROVED** [review]

**Pull request merged**